### PR TITLE
*: don't auto-resume for dump unit error

### LINKED ISSUE #837: do not "auto resume" for errors in dump unit
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   Run a task with "Couldn't acquire global lock"

2. What did you expect to see?

   the task keep paused

3. What did you see instead?

    still "auto resume".

    for errors in dump unit, most of them should not succeed after resumed, so no extra workload on upstream MySQL/MariaDB is better.

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v1.0.6
        ```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close #837

### What is changed and how it works?
move logic together, check PR does take effect

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to **master**

### CODE REVIEW

File: dm/worker/task_checker.go [hunk 1]
```diff
@@ -302,8 +300,10 @@ func (tsc *realTaskStatusChecker) getResumeStrategy(stStatus *pb.SubTaskStatus,
 	// TODO: use different strategies based on the error detail
 	for _, processErr := range stStatus.Result.Errors {
 		if !isResumableError(processErr) {
+			tsc.l.Info("error is not resumable", zap.Stringer("error", processErr))
```
Comment (csuzhangxc) [inline dm/worker/task_checker.go:303]: Maybe too many logs? but may not a problem.

Comment (lance6716) [reply dm/worker/task_checker.go:303]: used to check task_checker does find an error and decide not to resume it. going to enable this only in failpoint.

**Comment by lance6716** [comment]
@GMHDBJD PTAL

**Pull request merged**